eng/pipelines/build-test.yml

```diff
@@ -0,0 +1,123 @@
+              versionSpec: '18.13.0'
+          - script: |
+              npm ci
+            displayName: "Install packages"
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - script: |
+              npm run prettier
+            displayName: "Emitter format check"
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp/src/CADL.Extension/Emitter.Csharp
+          - script: 'dotnet pack src/AutoRest.CSharp/AutoRest.CSharp.csproj -o $(Build.ArtifactStagingDirectory) -warnaserror -c Release'
+            name: Package
+            displayName: 'Package'
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+            env:
+              DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+              DOTNET_CLI_TELEMETRY_OPTOUT: 1
+              DOTNET_MULTILEVEL_LOOKUP: 0
+          - pwsh: ./eng/SharedCodeCheck.ps1
+            displayName: "Check if shared source code is up-to-date"
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - task: Npm@1
+            displayName: 'Build CADL Ranch Mock Api project'
+            inputs:
+              command: custom
+              customCommand: run build
+              workingDir: $(Build.SourcesDirectory)/autorest.csharp/test/CadlRanchMockApis
+          - script: |
+              dotnet test AutoRest.CSharp.sln
+            displayName: "Test"
+            env:
+              DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+              DOTNET_CLI_TELEMETRY_OPTOUT: 1
+              DOTNET_MULTILEVEL_LOOKUP: 0
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - script: |
+              dotnet test AutoRest.CSharp.sln -c Release
+            displayName: "Test Release Configuration"
+            env:
+              DOTNET_SKIP_FIRST_TIME_EXPERIENCE: 1
+              DOTNET_CLI_TELEMETRY_OPTOUT: 1
+              DOTNET_MULTILEVEL_LOOKUP: 0
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - task: Npm@1
+            displayName: 'Build CADL csharp emitter'
+            inputs:
+              command: custom
+              customCommand: run build
+              workingDir: $(Build.SourcesDirectory)/autorest.csharp/src/CADL.Extension/Emitter.Csharp
+          - pwsh: ./eng/ExecuteCadlEmitterUnitTests.ps1
+            displayName: 'E2E Test for CADL emitter'
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - script: |
+              npm run test --prefix src/CADL.Extension/Emitter.Csharp
+            displayName: 'Unit Test'
+            workingDirectory: $(Build.SourcesDirectory)/autorest.csharp
+          - task: PublishBuildArtifacts@1
+            condition: succeededOrFailed()
+            displayName: 'Publish Artifacts'
+            inputs:
+              ArtifactName: packages
+          - ${{if and(ne(variables['Build.Reason'], 'PullRequest'), eq(variables['System.TeamProject'], 'internal'))}}:
```

Comment:
What is this publish doing? Should it also be moved to the publish.yml?

Reply:
In the `PublishRelease.ps1` will publish artifacts to github/npm, and it will also publish test reports. The logic is tightly coupled. I've created a follow-up issue to move publish artifacts steps to publish: https://github.com/Azure/autorest.csharp/issues/3169